tests/unit/statistical_testing/QuadraticTimeMMD_unittest.cc

```diff
@@ -354,7 +354,7 @@ TEST(QuadraticTimeMMD, perform_test_permutation_biased_full)
 	// assert against local machine computed result
 	mmd->set_statistic_type(ST_BIASED_FULL);
 	float64_t p_value=mmd->compute_p_value(mmd->compute_statistic());
-	EXPECT_NEAR(p_value, 0.0, 1E-10);
+	EXPECT_NEAR(p_value, 0.8, 1E-10);
```

Comment:
Yes absolutely!
What about creating a fixture class with a very simple and small fixed dataset, no randomness.
And then compute all the statistics and assert the results. Then put a link to a python notebook in there for reference. I did something like that here: https://github.com/karlnapf/shogun/blob/feature/kernel_exp_family/tests/unit/distribution/kernel_exp_family/impl/Nystrom_unittest.cc
and here
https://github.com/karlnapf/shogun/blob/feature/kernel_exp_family/tests/unit/distribution/kernel_exp_family/impl/DataFixture.h

As for the permutation test p-values. I would unit test the mechanics independently of ranodm data (if we have the statistics to be covered, we only need to test that the permutation test compares/computes the right things, i.e. test indices)

I think this would really benefit the maintainability of the framework!